pkg/kv/kvserver/replica_consistency_diff.go

```diff
@@ -36,46 +35,38 @@ type ReplicaSnapshotDiff struct {
 type ReplicaSnapshotDiffSlice []ReplicaSnapshotDiff
 
 // WriteTo writes a string representation of itself to the given writer.
-func (rsds ReplicaSnapshotDiffSlice) WriteTo(w io.Writer) (int64, error) {
-	n, err := w.Write([]byte("--- leaseholder\n+++ follower\n"))
-	if err != nil {
-		return 0, err
-	}
+func (rsds ReplicaSnapshotDiffSlice) WriteTo(buf *redact.StringBuilder) {
+	buf.Printf("--- leaseholder\n+++ follower\n")
 	for _, d := range rsds {
-		prefix := "+"
+		prefix := redact.Safe("+")
 		if d.LeaseHolder {
 			// Lease holder (RHS) has something follower (LHS) does not have.
-			prefix = "-"
+			prefix = redact.Safe("-")
 		}
 		ts := d.Timestamp
 		const format = `%s%d.%09d,%d %s
 %s    ts:%s
 %s    value:%s
 %s    raw mvcc_key/value: %x %x
 `
-		var prettyTime string
+		var prettyTime redact.SafeValue
 		if d.Timestamp == (hlc.Timestamp{}) {
-			prettyTime = "<zero>"
+			prettyTime = redact.Safe("<zero>")
 		} else {
-			prettyTime = d.Timestamp.GoTime().UTC().String()
+			prettyTime = redact.Safe(d.Timestamp.GoTime().UTC().String())
 		}
 		mvccKey := storage.MVCCKey{Key: d.Key, Timestamp: ts}
-		num, err := fmt.Fprintf(w, format,
-			prefix, ts.WallTime/1e9, ts.WallTime%1e9, ts.Logical, d.Key,
+		buf.Printf(format,
+			prefix, redact.Safe(ts.WallTime/1e9), redact.Safe(ts.WallTime%1e9), redact.Safe(ts.Logical), d.Key,
```

Comment:
are you sure these `redact.Safe` calls are necessary? IIRC numeric values are not redactable.